src/jdom/pipes.ts

```diff
@@ -80,7 +80,7 @@ export class InPipe extends JDClient {
     openCommand(cmd: number) {
         if (!this.isOpen)
             throwError("trying to access a closed pipe")
-        const b = pack("IIHH", [0, 0, this._port, 0])
+        const b = jdpack("i32 i32 u16 u16", [0, 0, this._port, 0])
```

Comment:
u32 u32 u16 u16

Reply:
doesn't matter, it's just 0s;
in fact it could be `"b[8] u16 u16", [deviceidbuffer, this._port, 0]`